src/Sulu/Bundle/AdminBundle/Resources/js/views/Form/toolbarActions/DeleteToolbarAction.js

```diff
@@ -55,45 +67,124 @@ export default class DeleteToolbarAction extends AbstractFormToolbarAction {
         super(resourceFormStore, form, router, locales, options, parentResourceStore);
     }
 
+    handleDeleteReferencedResourcesDialogCancel = () => {
+        this.closeDeleteReferencedResourceDialog();
+    };
+
+    @action handleDeleteReferencedResourcesDialogConfirm = () => {
+        this.delete(true);
+    };
+
+    @action closeDeleteReferencedResourceDialog = () => {
+        this.showDeleteReferencedResourcesDialog = false;
+        this.referencingResourcesData = null;
+    };
+
+    renderDeleteReferencedResourceDialog() {
+        if (!this.showDeleteReferencedResourcesDialog || this.referencingResourcesData === null) {
+            return null;
+        }
+
+        const {resource, referencingResources, referencingResourcesCount} = this.referencingResourcesData;
+
+        return (
+            <DeleteReferencedResourceDialog
+                allowDeletion={this.allowConflictDeletion}
+                loading={this.resourceFormStore.deleting}
+                onCancel={this.handleDeleteReferencedResourcesDialogCancel}
+                onConfirm={this.handleDeleteReferencedResourcesDialogConfirm}
+                referencingResources={referencingResources}
+                referencingResourcesCount={referencingResourcesCount}
+                resource={resource}
+            />
+        );
+    }
+
+    handleDeleteDependantsDialogFinish = () => {
+        this.delete();
+    };
+
+    handleDeleteDependantsDialogCancel = () => {
+        this.closeDeleteDependantsDialog();
+    };
+
+    handleDeleteDependantsDialogClose = () => {
+        this.closeDeleteDependantsDialog();
+    };
+
+    @action closeDeleteDependantsDialog = () => {
+        this.showDeleteDependantsDialog = false;
+        this.dependantResourcesData = null;
+    };
+
+    @computed get deleteDependantsDialogRequestOptions() {
+        const {locale, options: resourceFormStoreOptions = {}} = this.resourceFormStore;
+
+        const options = resourceFormStoreOptions;
+
+        if (locale) {
+            options.locale = locale.get();
+        }
+
+        return options;
+    }
+
+    renderDeleteDependantsDialog() {
+        if (!this.showDeleteDependantsDialog || this.dependantResourcesData === null) {
+            return null;
+        }
+
+        const {dependantResourcesCount, dependantResources} = this.dependantResourcesData;
+
+        return (
+            <DeleteDependantsDialog
```

Comment:
would call this `DeleteDependantResourcesDialog` to be consistent with `DeleteReferencedResourceDialog`